src/main/java/com/ufcg/es5/BackendComplexoEsportivoUFCG/application/court/service/CourtServiceImpl.java

```diff
@@ -34,6 +36,17 @@ public CourtResponseDto create(CourtSaveDto data) {
         return objectMapper.convertValue(court, CourtResponseDto.class);
     }
 
+    @Override
+    @Transactional
+    public CourtResponseDto update(CourtSaveDto data, Long id) throws SaceResourceNotFoundException {
+        Court court = repository.findById(id).orElseThrow(SaceResourceNotFoundException::new);
```

Comment:
trocar repository.findById por this.findById. E colocar a mensagem de erro adequada usando a classe de constantes